build/Dockerfile

```diff
@@ -1,40 +1,52 @@
 FROM debian:jessie
 MAINTAINER DST Academy <support@dst.academy>
 
-# Install dependencies.
-RUN apt-get update \
-	&& apt-get install -y --no-install-recommends \
-		ca-certificates \
-		curl \
-		lib32gcc1 \
-	&& apt-get clean \
-	&& rm -rf /var/lib/apt/lists/*
-
-# Install gosu.
-RUN gpg --keyserver pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 \
-	&& curl -o /usr/local/bin/gosu -fsSL "https://github.com/tianon/gosu/releases/download/1.9/gosu-$(dpkg --print-architecture)" \
-	&& curl -o /usr/local/bin/gosu.asc -fsSL "https://github.com/tianon/gosu/releases/download/1.9/gosu-$(dpkg --print-architecture).asc" \
-	&& gpg --verify /usr/local/bin/gosu.asc \
-	&& rm /usr/local/bin/gosu.asc \
-	&& chmod +x /usr/local/bin/gosu
+# Set build arguments.
+ARG GOSU_VERSION=1.9
 
 # Define environment variables.
 ENV STEAM_USER="steam" \
 	STEAM_HOME="/opt/steam"
 
-# Add steam user.
-RUN adduser --disabled-login --gecos "Steam" --home $STEAM_HOME $STEAM_USER
+# Add our user/group first to ensure their IDs get set consistently.
+RUN groupadd -r $STEAM_USER && useradd -rm -d $STEAM_HOME -g $STEAM_USER $STEAM_USER
+
+# Install everything.
+RUN set -x \
+
+	# Install dependencies.
+	&& apt-get update \
+	&& apt-get install -y --no-install-recommends wget ca-certificates lib32gcc1 \
 
-# Install SteamCMD.
-RUN curl -fsSL "https://cdn.steamstatic.com/client/installer/steamcmd_linux.tar.gz" | gosu $STEAM_USER tar -xzvC $STEAM_HOME \
+	# Install gosu.
+	&& wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$(dpkg --print-architecture)" \
+	&& wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$(dpkg --print-architecture).asc" \
+	&& export GNUPGHOME="$(mktemp -d)" \
+	&& gpg --keyserver ha.pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 \
+	&& gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu \
+	&& rm -r "$GNUPGHOME" /usr/local/bin/gosu.asc \
+	&& chmod +x /usr/local/bin/gosu \
+	&& gosu nobody true \
```

Comment:
Good question, I didn't check but it's used in the Dockerfiles by the guy who made gosu. I guess you're right; it ensures that gosu works.